src/IB/IBFEMethod.cpp

```diff
@@ -371,6 +371,13 @@ IBFEMethod::IBFEMethod(const std::string& object_name,
     : FEMechanicsBase(object_name, input_db, mesh, register_for_restart, restart_read_dirname, restart_restore_number),
       d_source_system_name(SOURCE_SYSTEM_NAME_VAL)
 {
+    // Check if the mesh dimention is compatable with the IBStategy
+    {
+        unsigned int mesh_dim = mesh->mesh_dimension();
+        unsigned int spatial_dim = mesh->spatial_dimension();
+        TBOX_ASSERT(mesh_dim == spatial_dim);
+    }
```

Comment:
```suggestion
    // Check if the mesh dimension is compatible with the IBStrategy
    TBOX_ASSERT(mesh->mesh_dimension() == mesh->spatial_dimension());
```
This is the correct fix but we can cut out several intermediate steps.